src/core/service.c

```diff
@@ -2882,9 +2938,21 @@ static int service_stop(Unit *u) {
 
 static int service_reload(Unit *u) {
         Service *s = ASSERT_PTR(SERVICE(u));
+        int r;
 
         assert(IN_SET(s->state, SERVICE_RUNNING, SERVICE_EXITED));
 
+        /* If we have confexts extensions, try to reload vpick'd confext
+         * extensions, which is particularly beneficial for notify-reload
+         * services that could potentially pick up a new version of its
+         * configuration.
+         */
+        if (service_should_reload_extensions(u)) {
+                r = service_reload_extensions(u);
+                if (r < 0)
+                        log_unit_error_errno(u, r, "Failed to reload confexts, ignoring: %m");
+        }
```

Comment:
Should this really be done here? This would create inconsistency between manual reloads and PropagatesReloadTo=?

Reply:
Sorry, are you referring to the reload invocation or the logging here?

Reply:
Assuming you mean the invocation, I've only envisioned the use case where we get here via a, say, dbus reload. It's fuzzier to me if a reload_by_notify should also take on this behavior.

Reply:
Yeah we should take the most conservative approach to start with, and then expand later if there's demand for it, so explicit reload is the way to go